src/main/webapp/view/discussion/discussion.js

```diff
@@ -103,57 +129,89 @@ async function fetchDiscussion() {
 }
 
 /**
- * Creates an element for displaying {@code comment}.
+ * Renders a comment and its replies, with a form to post a new reply.
  */
-function createComment(comment) {
-  const element = document.createElement('li');
-  element.setAttribute(ATTR_ID, comment.commentKey.id);
+class DiscussionComment extends HTMLElement {
+  /**
+   * @param comment The comment from the servlet that this element should
+   *     render.
+   */
+  constructor(comment) {
```

Comment:
This constructor is doing a lot. Can you add a high level comment explaining what it is doing? Not just the @param comment?

Reply:
Done!